### PR TITLE
[ASC-106] feat: new skeleton component

### PR DESCRIPTION
**Pull request by Harris-YWang**
## Description

Create a new root component for better loading states, with animation and different shapes.

btw, refine some other files for consistency.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):

Examples

Kapture 2020-11-24 at 09 49 40

Tile with Skeleton inside
Kapture 2020-11-24 at 09 47 52

### CODE REVIEW

**Review by Harris-YWang — COMMENTED** [review]
Design reviewed by @pedroscoulloupas

**Review by adityasabnis — APPROVED** [review]

**Review by vinteo — APPROVED** [review]
This is cool, but not sure about the skeleton name? I guess this was decided through the UI meeting?

**Comment by Harris-YWang** [comment]
> This is cool, but not sure about the skeleton name? I guess this was decided through the UI meeting?

yeah, we had a meeting about this one with @lteacher @superpower1, and we also found like `Material UI` and `Ant Design` UI libraries use `<Skeleton />` as the component name

**Review by lteacher — APPROVED** [review]

File: src/components/Skeleton/index.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  /**
+   *  oneOf: 'rect', 'circle', 'text'
+   */
+  variant: PropTypes.oneOf(['rect', 'circle', 'text']),
+  width: PropTypes.string,
+};
+
+Skeleton.defaultProps = {
+  animated: true,
+  variant: 'text',
```
Comment (superpower1) [inline src/components/Skeleton/index.jsx:43]: The default `text` variant is not used, remove it maybe?

File: src/components/Skeleton/index.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { expandDts } from '../../lib/utils';
+import './styles.scss';
+
+const Skeleton = ({ animated, className, dts, height, variant, width }) => {
+  const baseClass = 'aui--skeleton';
+
+  return (
+    <span
+      data-testid="skeleton-wrapper"
+      className={classnames(baseClass, className, { animated: animated }, variant)}
```
Comment (superpower1) [inline src/components/Skeleton/index.jsx:13]: I am not sure about using the variant directly as a className, maybe it's better to give it a prefix like `baseClass-variant` and also check for the variant value so if user put some random variant just ignore it.

File: src/components/Skeleton/index.jsx (hunk 1 above)
Comment (Harris-YWang) [reply src/components/Skeleton/index.jsx:43]: still need a default value for the default style, so I guess this one is okay. Or, any suggestion for the alternative choice?

File: src/components/Skeleton/index.jsx (hunk 2 above)
Comment (Harris-YWang) [reply src/components/Skeleton/index.jsx]: resolved

**Comment by vinteo** [comment]
Merging...

**Pull request merged**